iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -166,17 +166,36 @@ SharedMemoryObject::SharedMemoryObject(SharedMemory&& sharedMemory,
 {
 }
 
-void* SharedMemoryObject::allocate(const uint64_t size, const uint64_t alignment) noexcept
+cxx::expected<void*, SharedMemoryAllocationError> SharedMemoryObject::allocate(const uint64_t size,
+                                                                               const uint64_t alignment) noexcept
 {
-    return m_allocator.allocate(size, alignment);
+    if (size == 0)
+    {
+        IOX_LOG(WARN) << "Cannot allocate memory of size 0.";
+        return cxx::error<SharedMemoryAllocationError>(SharedMemoryAllocationError::REQUESTED_ZERO_SIZED_MEMORY);
```

Comment:
IMO this should be an assert like check, it is a violation of the precondition size>0.

Precondition violations indicate bugs and should never lead to error propagation but (as graceful as possible) termination and a log entry indicating the bug.

We can solve this later while reworking how we deal with errors though.